src/map/battleground.cpp

```diff
@@ -222,59 +621,863 @@ int bg_team_get_id(struct block_list *bl)
+
+		if (!r)
+			queue->teama_members.push_back(sd);
+		else
+			queue->teamb_members.push_back(sd);
+
+		sd->bg_queue = queue;
+		bg_queues.insert(bg_queues.begin(), queue);
+		return BG_APPLY_ACCEPT;
+	} else {
+		r = rnd() % 2 != 0;
+
+		for (const auto &it : bg_queues) {
+			try {
+				queue = it;
+			} catch (std::out_of_range &) {
+				continue;
+			}
+
+			if (queue->in_ready_state)
+				continue;
+			if (!r) {
+				if (queue->teama_members.size() != queue->required_players) {
+					sd->bg_queue = queue;
+					queue->teama_members.push_back(sd);
+
+					if (queue->teama_members.size() == bg->required_players && queue->teamb_members.size() == bg->required_players) // Enough players have joined
+						bg_queue_on_ready(name, queue);
+					return BG_APPLY_ACCEPT;
+				} else if (queue->teamb_members.size() != queue->required_players) {
+					sd->bg_queue = queue;
+					queue->teamb_members.push_back(sd);
+
+					if (queue->teama_members.size() == bg->required_players && queue->teamb_members.size() == bg->required_players) // Enough players have joined
+						bg_queue_on_ready(name, queue);
+					return BG_APPLY_ACCEPT;
+				}
+			} else {
+				if (queue->teamb_members.size() != queue->required_players) {
+					sd->bg_queue = queue;
+					queue->teamb_members.push_back(sd);
+
+					if (queue->teama_members.size() == bg->required_players && queue->teamb_members.size() == bg->required_players) // Enough players have joined
+						bg_queue_on_ready(name, queue);
+					return BG_APPLY_ACCEPT;
+				} else if (queue->teama_members.size() != queue->required_players) {
+					sd->bg_queue = queue;
+					queue->teama_members.push_back(sd);
+
+					if (queue->teama_members.size() == bg->required_players && queue->teamb_members.size() == bg->required_players) // Enough players have joined
+						bg_queue_on_ready(name, queue);
+					return BG_APPLY_ACCEPT;
+				}
+			}
+		}
+	}
+
+	queue = bg_queue_create(bg);
+	r = rnd() % 2 != 0;
+
+	if (!r)
+		queue->teama_members.push_back(sd);
+	else
+		queue->teamb_members.push_back(sd);
+
+	sd->bg_queue = queue;
+	bg_queues.insert(bg_queues.begin(), queue);
+	return BG_APPLY_ACCEPT;
+}
+
+/**
+ * Join a party onto the same side of a Battleground
+ * @param name: Battleground name
+ * @param sd: Player who requested to join the battlegrounds
+ * @return @see e_bg_queue_apply_ack
+ */
+enum e_bg_queue_apply_ack bg_queue_join_party(char *name, struct map_session_data *sd)
+{
+	struct party_data *p = party_search(sd->status.party_id);
+
+	if (!p)
+		return BG_APPLY_INVALID_APP; // Someone has bypassed the client check for being in a party
+
+	uint32 pl_cid = 0;
+
+	for (const auto &it : p->party.member) {
+		if (it.leader) {
+			pl_cid = it.char_id;
+			break;
+		}
+	}
+
+	if (sd->status.char_id != pl_cid)
+		return BG_APPLY_PARTYGUILD_LEADER; // Not the party leader
+	else {
+		int p_online = 0;
+
+		if (auto bg = bg_search_name(name).lock()) {
+			for (const auto &it : p->party.member) {
```

Comment:
This for loop can be easily replaced with an `std::count_if`
```suggestion
			int player_online = std::count_if(p->party.member.begin(), p->party.member.end(), [](const auto& member) { return member.online; })
```

Reply:
This won't work since party member data is not a vector or something that contains iterator form (begin, end). This could definitely be something refactored in the future when party is converted to C++.